dali/operators/coord/coord_flip.cc

```diff
@@ -0,0 +1,105 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/coord/coord_flip.h"
+
+namespace dali {
+
+DALI_SCHEMA(CoordFlip)
+    .DocStr(
+        R"code(Transforms coordinates so that they are flipped (point reflected) with respect
+to a center point.)code")
+    .NumInput(1)
+    .NumOutput(1)
+    .AddOptionalArg<TensorLayout>(
+      "layout",
+      R"code(Determines the layout of the coordinates. Possible values are:
+
+  - ``x`` (horizontal position), 
+  - ``y`` (vertical position), 
+  - ``z`` (depthwise position),
+
+Note: If left empty, ``"x"``, ``"xy"`` or ``"xyz"`` will be assumed, depending on the number of dimensions.
+)code",
+      TensorLayout{""})
+    .AddOptionalArg("flip_x", R"code(Flip horizontal (x) dimension.)code", 1, true)
+    .AddOptionalArg("flip_y", R"code(Flip vertical (y) dimension.)code", 0, true)
+    .AddOptionalArg("flip_z", R"code(Flip depthwise (z) dimension.)code", 0, true)
+    .AddOptionalArg("center_x", R"code(Flip center on horizontal dimension.)code", 0.5f, true)
+    .AddOptionalArg("center_y", R"code(Flip center on vertical dimension.)code", 0.5f, true)
+    .AddOptionalArg("center_z", R"code(Flip center on depthwise dimension.)code", 0.5f, true);
+
+
+class CoordFlipCPU : public CoordFlip<CPUBackend> {
+ public:
+  explicit CoordFlipCPU(const OpSpec &spec)
+      : CoordFlip<CPUBackend>(spec) {}
+
+  ~CoordFlipCPU() override = default;
+  DISABLE_COPY_MOVE_ASSIGN(CoordFlipCPU);
+
+  void RunImpl(workspace_t<CPUBackend> &ws) override;
+
+  USE_OPERATOR_MEMBERS();
+  using Operator<CPUBackend>::RunImpl;
+  using CoordFlip<CPUBackend>::layout_;
+};
+
+void CoordFlipCPU::RunImpl(workspace_t<CPUBackend> &ws) {
+  const auto &input = ws.InputRef<CPUBackend>(0);
+  auto &output = ws.OutputRef<CPUBackend>(0);
+  auto &thread_pool = ws.GetThreadPool();
+
+  int x_dim = layout_.find('x');
+  DALI_ENFORCE(x_dim >= 0, "Dimension \"x\" not found in the layout");
+
+  int y_dim = layout_.find('y');
+  if (ndim_ > 1)
+    DALI_ENFORCE(y_dim >= 0, "Dimension \"y\" not found in the layout");
+
+  int z_dim = layout_.find('z');
+  if (ndim_ > 2)
+    DALI_ENFORCE(z_dim >= 0, "Dimension \"z\" not found in the layout");
+
+  for (int sample_id = 0; sample_id < batch_size_; sample_id++) {
+    std::array<bool, 3> flip_dim = {false, false, false};
+    flip_dim[x_dim] = spec_.GetArgument<int>("flip_x", &ws, sample_id);
+    flip_dim[y_dim] = spec_.GetArgument<int>("flip_y", &ws, sample_id);
+    flip_dim[z_dim] = spec_.GetArgument<int>("flip_z", &ws, sample_id);
```

Comment:
y_dim and z_dim can be -1 if they are not in layout (1D and 2D case). Same below.